crates/ty_python_semantic/src/types/class_base.rs

```diff
@@ -161,7 +161,8 @@ impl<'db> ClassBase<'db> {
             | Type::AlwaysFalsy
             | Type::AlwaysTruthy
             | Type::TypeIs(_)
-            | Type::TypedDict(_) => None,
+            | Type::TypedDict(_)
+            | Type::NewTypeInstance(_) => None,
```

Comment:
I'd feel slightly safer here if we delegated:

```suggestion
            | Type::TypedDict(_) => None,
            
            Type::NewTypeInstance(newtype) => {
                ClassBase::try_from_type(db, Type::instance(db, newtype.base_class_type(db)), subclass)
            }
```

It's unlikely that we'll ever accept any `NominalInstance` types as base classes, but there _is_ a TODO above the `NominalInstance` branch above which could _plausibly_ be fixed, and it seems important to keep the behaviour for a given `NewType` here consistent with the behaviour of that `NewType`'s nominal-instance supertype